packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/handles/ConnectionCreationHandles.tsx

```diff
@@ -105,6 +105,18 @@ export const ConnectionCreationHandles = ({ nodeId }: ConnectionCreationHandlesP
     }));
   };
 
+  const connectionNodeId = useStore((state) => state.connectionNodeId);
+  const isConnecting = !!connectionNodeId;
+  const { connection } = useConnector();
+  useEffect(() => {
+    if (!isConnecting && !connection) {
+      setState((prevState) => ({
+        ...prevState,
+        isMouseDown: null,
+      }));
+    }
+  }, [isConnecting, connection]);
```

Comment:
...Then adapt this `useEffect` to rely on `connectionInProgress()` from the hook.

Reply:
done